npm/pkg/dataplane/policies/chain-management_linux.go

```diff
@@ -64,7 +66,25 @@ type staleChains struct {
 }
 
 func newStaleChains() *staleChains {
-	return &staleChains{make(map[string]struct{})}
+	return &staleChains{
+		chainsToCleanup: make(map[string]struct{}),
+	}
+}
+
+// forceLock stops reconciling if it is running, and then locks the reconcileManager
+func (rm *reconcileManager) forceLock() {
+	rm.releaseLockSignal <- struct{}{}
+	rm.Lock()
+}
+
+// forceUnlock makes sure that the releaseLockSignal channel is empty (in case reconciling
+// wasn't running when forceLock was called), and then unlocks the reconcileManager.
+func (rm *reconcileManager) forceUnlock() {
+	select {
+	case <-rm.releaseLockSignal:
+	default:
+	}
```

Comment:
You dont need a select block when using just one channel/case.
```suggestion
	<-rm.releaseLockSignal:
```

Reply:
The idea of this code is to empty the signal, which might be empty already. This is where the default case comes in because then we might come into deadlock.